tests/test_executors.py

```diff
@@ -71,47 +80,35 @@ class QuantumExecutorMock(QuantumExecutor):
         Whether the mock should use multiprocessing or not.
     """
 
-    def __init__(self, mp=False):
-        self.quanta = []
+    def __init__(self, mp: bool = False):
+        self.quanta: list[QuantumMock] | ListProxy = []
         if mp:
             # in multiprocess mode use shared list
             manager = Manager()
             self.quanta = manager.list()
-        self.report = None
-        self._execute_called = False
 
-    def execute(self, task_node, /, quantum):
+    def execute(  # type: ignore[override]
+        self, task_node: TaskNodeMock, /, quantum: QuantumMock  # type: ignore[override]
+    ) -> tuple[QuantumMock, QuantumReport | None]:
```

Comment:
This commit still mixes typing and report changes. This is exactly why it's best to commit them separately in the first place.

Unfortunately, I don't see an easy way to disentangle the commits now -- both sets of changes need to modify `test_executors.py`.

Reply:
True, and I do not want to spend much time on trying to perfect it.